homeassistant/components/alarmdecoder/alarm_control_panel.py

```diff
@@ -39,14 +43,57 @@
 ALARM_KEYPRESS_SCHEMA = vol.Schema({vol.Required(ATTR_KEYPRESS): cv.string})
 
 
+def get_arm_sequences(panel_brand, code_arm_required, alt_night_mode):
```

Comment:
All of this should be moved to the library. The library should handle brand identification and expose friendly named methods for the library user. The panel sent code details should be abstracted by the library.

Reply:
@MartinHjelmare, I'm not sure the AlarmDecoder author would agree with that. AlarmDecoder exposes the `send()` method which is responsible for delivering keypresses to Honeywell/DSC alarm panels. _Which_ keypresses are sent to those panels is ultimately left up to AlarmDecoder users.

Personally, I think HomeAssistant is the correct spot for these sequences. However, if there's something incorrect about my implementation, I'm happy to move things around for better organization or to adhere to any code rules set by Home Assistant.

If there's no room for these sequences in Home Assistant, I think the only other option would be to create some kind of _homeassistant-alarmdecoder-connector_ library which seems excessive and not worth the effort.

Another noteworthy point is that AlarmDecoder is a pretty mature product at this point and therefore this collection of arming sequences is unlikely to grow in the future. I thoroughly documented the key sequences in [this documentation PR](https://github.com/home-assistant/home-assistant.io/pull/13494) and did my best to make their purpose explicit in the code itself. This should help developers down the road for when my time as an AlarmDecoder codeowner is up.

Please let me know your thoughts. Based on #35728 and a recent email I received from another HomeAssistant/AlarmDecoder user, I know this would be a welcomed feature to other HomeAssistant/AlarmDecoder users.

Reply:
If a new 3rd part library is required, that's ok. We won't accept more device specific interface code in home assistant for this integration.

Reply:
Okay. Feel free to close this PR if it isn't going to be accepted then. I'll look into creating a 3rd party library at some point in the future.